engine/classes/ElggEntity.php

```diff
@@ -1068,6 +1068,29 @@ public function canEdit($user_guid = 0) {
 	}
 
 	/**
+	 * Can a user delete this entity?
+	 *
+	 * @tip Can be overridden by registering for the permissions_check:delete plugin hook.
+	 *
+	 * @param int $user_guid The user GUID, optionally (default: logged in user)
+	 *
+	 * @return bool Whether this entity is deletable by the given user.
+	 * @see elgg_set_ignore_access()
+	 */
+	public function canDelete($user_guid = 0) {
+		$user_guid = (int)$user_guid;
+		$user = get_entity($user_guid);
+		if (!$user) {
```

Comment:
If a non-zero $user_guid is given, use `_elgg_services()->entityTable->get($user_guid, 'user')` and if it returns false, we should immediately return false. It means the user doesn't exist or the session can't see the user. In either case we can't continue.